Src/FluentAssertions/Equivalency/Steps/EnumerableEquivalencyValidator.cs

```diff
@@ -34,6 +34,8 @@ public EnumerableEquivalencyValidator(IEquivalencyValidator parent, IEquivalency
 
     public void Execute<T>(object[] subject, T[] expectation)
     {
+        AssertionScope.Current.BecauseOf(context.Reason);
```

Comment:
🤔 I think you need to solve this problem at a different place. When a new nested `AssertionScope` is created, such as [here](https://github.com/fluentassertions/fluentassertions/blob/a832de3ed8987188d13cf4ae33bf4413e6137f9a/Src/FluentAssertions/Equivalency/Steps/GenericDictionaryEquivalencyStep.cs#L157), it should automatically pick up the _reason_ from the parent. This should be fixed [here](https://github.com/fluentassertions/fluentassertions/blob/de9fa4d63ffad9444657558ae690ccbcc789dfc2/Src/FluentAssertions/Execution/AssertionScope.cs#L106).

Reply:
Thank you at pointing me at the correct location ♥️